dfsclient/src/main/java/org/hpccsystems/dfs/client/AvroFieldTranslator.java

```diff
@@ -0,0 +1,1036 @@
+                        map.put(key,val);
+                    }
+
+                    return map;
+                }
+                else
+                {
+                    throw new Exception("AvroFieldTranslator toAvro: Map: Unexpected value type: " + field.getClass());
+                }
+            }
+            case NULL:
+            {
+                if (fd.getFieldType() != FieldType.BOOLEAN)
+                {
+                    throw new Exception("AvroFieldTranslator toAvro: incompatible avro schema type: " 
+                                        + schema.getType() + "  to hpcc field type: " + fd.getFieldType());
+                }
+
+                if (field instanceof Boolean)
+                {
+                    return field;
+                }
+                else
+                {
+                    throw new Exception("AvroFieldTranslator toAvro: Null: Unexpected value type: " + field.getClass());
+                }
+            }
+            case RECORD:
+            {
+                if (fd.getFieldType() != FieldType.RECORD)
+                {
+                    throw new Exception("AvroFieldTranslator toAvro: incompatible avro schema type: " 
+                                        + schema.getType() + "  to hpcc field type: " + fd.getFieldType());
+                }
+
+                if ((field instanceof HPCCRecord) == false)
+                {
+                    throw new Exception("AvroFieldTranslator toAvro: Record: Unexpected value type: " + field.getClass());
+                }
+
+                HPCCRecord record = (HPCCRecord) field;
+
+                // Translate child fields
+                List<Schema.Field> schemaFields = schema.getFields();
+                ArrayList<Object> translatedFields = new ArrayList<Object>();
+                
+                int fieldIndex = 0;
+                for (Schema.Field schemaField : schemaFields)
+                {
+                    Object childField = record.getField(fieldIndex);
+                    if (schemaField.schema().getType() == Schema.Type.UNION)
+                    {
+                        Object outField = hpccUnionToAvro(schemaField.schema(), fd.getDef(fieldIndex), childField, false);
+                        translatedFields.add(outField);
+                    }
+                    else
+                    {
+                        Object outField = toAvro(schemaField.schema(), fd.getDef(fieldIndex), childField);
+                        translatedFields.add(outField);
+                    }
+
+                    fieldIndex++;
+                }
+
+                // If we have additional child fields then null bit fields were created
+                int numNullBitFields = 0;
+                if (schemaFields.size() == fd.getNumDefs())
+                {
+                    numNullBitFields = (translatedFields.size() + 63) / 64;
```

Comment:
what is the significance of 63 and 64? can we assign those values to self-documenting static/final vars?